src/components/search/SearchResults.tsx

```diff
@@ -100,28 +93,72 @@ const InnerSearchResultList = <T extends SearchResultsType>(
     return router.query[param]
   }
 
+  const getSearchHits = createFetchDataFn<GetSearchResults_searchQuery_hits[]>([])({
+    // Need to define this since it can't be undefined
+    chain: async ({ api }: { api: SubsocialApi }) => {
+      return []
+    },
+    squid: async (
+      {
+        tab,
+        tags,
+        query,
+        spaceId,
+        offset,
+        limit,
+      }: {
+        tab: ElasticSearchIndexType[]
+        tags: string[]
+        query: string
+        spaceId: string
+        offset: number
+        limit: number
+      },
+      client,
+    ) => {
+      const hits = await searchResults(client, {
+        indexes: tab,
+        spaceId,
+        q: query,
+        tags,
+        offset,
+        limit,
+      })
+
+      const filteredHits: GetSearchResults_searchQuery_hits[] = hits.filter(
+        hit => hit !== null,
+      ) as GetSearchResults_searchQuery_hits[]
+
+      return filteredHits
+    },
+  })
+
   const querySearch: InnerLoadMoreFn<SearchResultsType> = async (page, size) => {
-    const tab = getReqParam('tab') as ElasticIndexTypes[]
+    const tab = getReqParam('tab') as ElasticSearchIndexType[]
     const query = getReqParam('q') as string
     const tags = getReqParam('tags') as string[]
     const spaceId = getReqParam('spaceId') as string
     const offset = (page - 1) * size
 
-    const res = await queryElasticSearch({
-      indexes: tab || AllTabKey,
-      spaceId,
-      q: query,
-      tags,
-      offset,
-      limit: size,
+    const hits = await getSearchHits(DataSourceTypes.SQUID, {
```

Comment:
so if above change is implemented, you don't need to specify the data source and just call that function

Reply:
@teodorus-nathaniel implemented in [d88b004](https://github.com/dappforce/polkaverse/pull/87/commits/d88b0047f96955438d5d0f0204108d1617d72b8c)